packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -79,49 +80,64 @@ export class MonacoOutlineContribution implements FrontendApplicationContributio
             return;
         }
         this.toDisposeOnClose.push(this.toDisposeOnEditor);
+        this.toDisposeOnEditor.push(Disposable.create(() => this.roots = undefined));
         const editor = this.editorManager.currentEditor;
         if (editor) {
             const model = MonacoEditor.get(editor)!.getControl().getModel();
-            this.toDisposeOnEditor.push(model.onDidChangeContent(() => this.updateOutline()));
+            this.toDisposeOnEditor.push(model.onDidChangeContent(() => {
+                this.roots = undefined; // Invalidate the previously resolved roots.
+                this.updateOutline();
+            }));
+            this.toDisposeOnEditor.push(editor.editor.onSelectionChanged(selection => this.updateOutline(selection)));
```

Comment:
We should ignore selection changes if they are triggered by the monaco outline contribution, otherwise, it triggers flickering. You can reproduce it by navigating in `DockPanelRenderer` in the outline, try to navigate down from `constructor`.

Reply:
How can I reproduce this flickering? Perhaps, I do not understand what you mean by flickering. I selected the `constructor` and did the navigation with the down key. See the attached screencast:

![screencast 2018-09-19 17-38-14](https://user-images.githubusercontent.com/1405703/45764595-5928cc00-bc33-11e8-8222-7945817b60c2.gif)

Reply:
I mean it does not select `tabBarRendererFactory` in the outline when you press down from `constructor`.